tests/test_buffer.py

```diff
@@ -598,6 +598,52 @@ def test_SimpleArray_broadcast_slice_ghost_md(self):
                 for k in range(4):
                     self.assertEqual(ndarr2[i, j, k], sarr[i - G, j, k])
 
+    def test_SimpleArray_casting(self):
+        # first check the caster works if the argument is exact the same type
+        array_float64 = modmesh.SimpleArrayFloat64((2, 3, 4))
+        self.assertEqual(modmesh.testhelper.TestSimpleArrayHelper.
+                         test_load_arrayfloat64_from_arrayplex(array_float64),
+                         True)
+
+        # init arrayplex
+        arrayplex_int32 = modmesh.SimpleArray((2, 3, 4), dtype="int32")
+        arrayplex_uint64 = modmesh.SimpleArray((2, 3, 4), dtype="uint64")
+        arrayplex_float64 = modmesh.SimpleArray((2, 3, 4), dtype="float64")
+
+        # check the type is the same with different data types
```

Comment:
Please also assert the type name string.  It will make the test code more descriptive.

Reply:
done